lib/miq_expression.rb

```diff
@@ -1623,27 +1630,34 @@ def self.ruby_for_date_compare(col_ruby, col_type, tz, op1, val1, op2 = nil, val
 
   def to_arel(exp, tz)
     operator = exp.keys.first
-
     field = Field.parse(exp[operator]["field"]) if exp[operator].kind_of?(Hash) && exp[operator]["field"]
+    if ALLOWED_OPERATORS_EXPRESSION_AS_VALUE.include?(operator) && Field.valid_field?(exp[operator]["value"])
+      value_field = Field.parse(exp[operator]["value"]).arel_attribute if exp[operator].kind_of?(Hash) && exp[operator]["value"]
```

Comment:
@alongoldboim aah, I now understand this.

Could you move `value_field` up right next to `field`?

```ruby
value_field = exp[operator]["value"] if exp[operator].kind_of?(Hash)
```

Then the purpose of the `Field.valid_field?` will be there to swap `value_field` with an `arel_attribute`